src/hd.rs

```diff
@@ -21,33 +21,31 @@ lazy_static! {
 
 #[derive(Debug)]
 pub struct HDWatcher {
-    wallets: HashMap<Fingerprint, HDWallet>,
+    wallets: HashMap<DescriptorChecksum, HDWallet>,
 }
 
 impl HDWatcher {
     pub fn new(wallets: Vec<HDWallet>) -> Self {
         HDWatcher {
             wallets: wallets
                 .into_iter()
-                .map(|wallet| (wallet.master.fingerprint(), wallet))
+                .map(|wallet| (wallet.descriptor.checksum(), wallet))
                 .collect(),
         }
-        // XXX indexing by the fingerprint prevents using the same underlying public key with
-        // different script types ([xyz]pub), they will have the same fingerprint and conflict.
     }
 
-    pub fn wallets(&self) -> &HashMap<Fingerprint, HDWallet> {
+    pub fn wallets(&self) -> &HashMap<DescriptorChecksum, HDWallet> {
         &self.wallets
     }
 
-    pub fn get(&self, fingerprint: Fingerprint) -> Option<&HDWallet> {
-        self.wallets.get(&fingerprint)
+    pub fn get(&self, desc_check: DescriptorChecksum) -> Option<&HDWallet> {
```

Comment:
Open to suggestions naming this `DescriptorChecksum` struct and convention for naming the instances (`desc_checks`)

Reply:
Maybe shorten to `DescrChecksum` and `descr_cs`?